include/envoy/stream_info/stream_info.h

```diff
@@ -64,6 +65,22 @@ enum ResponseFlag {
   LastFlag = StreamIdleTimeout
 };
 
+/**
+ * Constants for the response code details field of StreamInfo.
+ *
+ * These provide details about the stream state such as whether the
+ * response is from the upstream or from envoy (in case of a local reply).
+ * Custom extensions can define additional values provided they are appropriately
+ * scoped to avoid collisions.
+ */
+struct ResponseCodeDetailValues {
+  const std::string SET_BY_UPSTREAM = "set_by_upstream";
```

Comment:
nit: "set_by_upstream" is a bit vague to me in terms of what it means given we have lots of info in stream info. Perhaps "via_upstream" ? Any other ideas? Also nit, `SetByUpstream` (trying to avoid capitals for new constants).

Reply:
"proxied_from_upstream" or "forwarded_from_upstream"? "via_upstream" is fine too and shorter so I'll go with that :)